src/main/java/favourite/use_case/UserStringFactory.java

```diff
@@ -0,0 +1,22 @@
+package favourite.use_case;
+
+import entities.User;
+
+import java.util.ArrayList;
+
+/**
+ * UserStringFactory in the use case layer implements UserWriterInterface.
+ * It converts User objects into a format that the database can save.
+ */
+public class UserStringFactory implements UserWriterInterface{
```

Comment:
I don't think this needs the interface, as it should be called by the interactor, similarly to above with the post case